### ISSUE TITLE
Q: merging sparse tensors - is order important

### ISSUE THREAD

**Issue by martinrebane** [mit-han-lab/torchsparse#124]
Hi! I need to merge/concat two sparse tensors `a` and `b`, but not feature-wise as in `torchsparse/torchsparse/operators.py` where `a.F` and `b.F` are concatenated by dim=1 and a.C kept as coordinate tensor. I need to concat them tensor-wise on dim=0. Such that all features from `a` and `b` are kept, but merged into a single tensor, the same for coords (.C). So if tensor `a` had `x` elements and tensor `b` had `y` elements then the resulting tensor would have `x+y` elements. Looking at the code it is probably as trivial as 
```
new.F = torch.cat((a.F,b.F))
new.C = torch.cat((a.C,b.C))
```
But just to be on the safe side I have a few questions:
1) Elements in Sparse tensor seem to be ordered, does it matter? Should I reorder the new tensor?
2) Should I just reset .cmaps and .kmaps or merge them too?
3) is there anything else to keep in mind?

Thanks a million! :)

**Comment by zhijian-liu** [comment]
I guess what you are asking for is to stack over the batch dimension. The implementation is mostly correct. However, you need to keep in mind that the fourth dimension is `.C` is the batch index (zero-indexed). Thus, you need to add an offset (i.e., the batch size of `a`) to the batch index in `b.C`.

**Comment by martinrebane** [comment]
Thanks for the clarification! The `sparse_quantize` example in the README removes all duplicate values on same coordinates in a voxel grid. Is this a requirement (i.e., sparse operations would fail otherwise) or just to speed things up?

**Comment by Barryfxx** [comment]
Hi, I encountered this error: 'SparseTensor' object has no attribute 'kmaps', while running:
    new_tensor = SparseTensor(inserted_feat, x.C, x.s)
    new_tensor._caches.cmaps = x._caches.cmaps
    new_tensor.kmaps = x.kmaps
version: torchsparse 2.1.0+torch113cu117, how can I fix it?
